core/unit_test/TestHalfOperators.hpp

```diff
@@ -882,14 +882,17 @@ struct Functor_TestHalfOperators {
     }
 
     // TODO: figure out why operator{!,&&,||} are returning __nv_bool
-    actual_lhs(NEG)   = static_cast<double>(!h_lhs);
-    expected_lhs(NEG) = !d_lhs;
+    actual_lhs(NEG)   = static_cast<double>(!static_cast<bool>(h_lhs));
```

Comment:
this is wrong: we actually WANT to test implicit conversion for various usage scenarios